client/python/examples/async_logging.py

```diff
@@ -0,0 +1,210 @@
+
+    This is a blocking call, so it will never return.
+
+    If there have been more than 10 failed connections, then it will exit
+    """
+    attempts = 0
+
+    # Continuely try and reconnected to the job set
+    while True:
+        try:
+            event_stream = client.get_job_events_stream(
+                queue=queue, job_set_id=job_set_id
+            )
+
+            # For each event, check if it is one we are interested in
+            # and print out the message if it is
+            for event in event_stream:
+
+                event = client.unmarshal_event_response(event)
+
+                print(f"Job {event.message.job_id} - {event.type}")
+
+        # Handle the error we expect to maybe occur
+        except grpc.RpcError as e:
+            if e.code() != grpc.StatusCode.NOT_FOUND:
+                print("Unexpected error:", e)
+                exit()
+
+        attempts += 1
+        time.sleep(1)
+
+        if attempts > 10:
+            print("Exiting - too many attempts")
+            exit()
+
+
+def watch_queue(client, queue):
+    """
+    Watch the queue and any changes that occur
+
+    Print out the changes
+    """
+
+    last_info1 = None
+    last_info2 = None
+
+    # Continously featch the queue infomation
+    while True:
+        info1 = client.get_queue_info(name=queue)
+        info2 = client.get_queue(name=queue)
+
+        if last_info1 is None:
+            last_info1 = info1
+            last_info2 = info2
+
+        # If there is a change in the priority factor, print it out
+        if last_info2.priority_factor != info2.priority_factor:
+            print(
+                f"Priority factor changed from {last_info2.priority_factor} "
+                f"to {info2.priority_factor}"
+            )
+
+        # If there is a change in the number of jobs, print it out
+        for new, old in zip(info1.active_job_sets, last_info1.active_job_sets):
+            if new.leased_jobs != old.leased_jobs:
+                print(
+                    f"Leased jobs changed from {old.leased_jobs} to {new.leased_jobs}"
+                )
+
+        last_info1 = info1
+        last_info2 = info2
+
+        # So that we don't spam the server
+        time.sleep(0.2)
+
+
+def workflow(client, queue, job_set_id):
+    """
+    Example workflow for the async logging example
+    """
+
+    # Handle if the queue already exists
+    try:
+        client.create_queue(name=queue, priority_factor=1)
+
+    # Handle the error we expect to maybe occur
+    except grpc.RpcError as e:
+        code = e.code()
+        if code == grpc.StatusCode.ALREADY_EXISTS:
+            print(f"Queue {queue} already exists")
+            client.update_queue(name=queue, priority_factor=1)
+        else:
+            raise e
+
+    # Time for the watcher to pick it up
+    time.sleep(1)
+
+    # Some different commands for logging to detect
+    client.update_queue(name=queue, priority_factor=2)
```

Comment:
This seems to make the reprioritize_jobs does nothing.  You update queue with 2 and then reprioritize with 2 again

Reply:
Oh. I thought that queues had priority and then job_sets had their own seperate priority?

Reply:
Yea you know this better than I.  You are right.